SynthSeg/training_with_tfrecords.py

```diff
@@ -183,21 +228,87 @@ def load_model(
                 for (key, value) in getmembers(nrn_layers, isclass)
                 if key != "Layer"
             }
+
             custom_objects = {
                 **custom_l2i,
                 **custom_nrn,
                 "tf": tf,
                 "keras": tf.keras,
                 "loss": IdentityLoss().loss,
+                "DiceLoss": DiceLoss
             }
             model = tf.keras.models.load_model(
                 checkpoint, custom_objects=custom_objects
             )
+            init_batch_idx = model.optimizer.iterations.numpy()
+
             is_compiled = True
         else:
             model.load_weights(checkpoint, by_name=True)
-
-    return model, is_compiled, init_epoch
+            print("Loading weights only")
+
+    return model, is_compiled, init_epoch, init_batch_idx
+
+
+
+class TimingCallback(tf.keras.callbacks.Callback):
```

Comment:
I don't think we need this callback, I would prefer a more lightweight PR where we address the two main issues: container and resuming

Reply:
removed:)